### PR TITLE
feat(helm): support `--kube-version`, `--api-versions`, and `--namespace` (#19325)

### LINKED ISSUE #19325: Add `spec.source.helm` fields for `--kube-version`, `--api-versions`, and `--namespace`
# Summary

We should add fields in `spec.source.helm` for the Helm CLI's `--kube-version`, `--api-versions`, and `--namespace` flags.

Today, all three of those parameters are inferred from the `spec.destination`. The Kube version and API versions are pulled from the destination cluster. And the `--namespace` field is set to the `spec.destination.namespace` field (if present).

# Motivation

Users should have a way to make these dynamic values static so that manifest hydration is predictable/deterministic.

Most users likely will be fine with the default values dynamically loaded at manifest hydration time. Some may want the additional control.

Adding these fields will help provide a full feature set to the [manifest hydrator](https://github.com/argoproj/argo-cd/pull/17755/files) which is meant to generate manifests deterministically and should avoid accepting runtime-generated variables.

# Proposal

We should add these fields to the Application CRD:

```yaml
spec:
  source:
    helm:
      # You can specify the Kubernetes API version to pass to Helm when templating manifests. By default, Argo CD uses
      # the Kubernetes version of the target cluster.
      kubeVersion: v1.30

      # You can specify the Kubernetes resource API versions to pass to Helm when templating manifests. By default, Argo
      # CD uses the API versions of the target cluster. The format is [group/]version/kind.
      apiVersions:
        - traefik.io/v1alpha1/TLSOption
        - v1/Service

      # Optional namespace to template with. If left empty, defaults to the app's destination namespace.
      namespace: custom-namespace
```

When one is specified, the corresponding dynamically-loaded value should be ignored in favor of the hard-coded value.

We should provide a CLI interface to set these values.

We should not provide a UI interface to set these values. This is more of a power-user feature set, and it would likely be more confusing than helpful to UI users.

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Closes #19325

### CODE REVIEW

**Review by agaudreault — COMMENTED** [review]
LGTM

File: controller/state.go [hunk 1]
```diff
@@ -275,6 +283,20 @@ func (m *appStateManager) GetRepoObjs(app *v1alpha1.Application, sources []v1alp
 	return targetObjs, manifestInfos, nil
 }
 
+// getServerAndApiVersions will return the server version and the API versions for the given application. If either the
+// server version or the API versions are not provided in the application spec, we will use the value from the cluster.
+func getServerAndApiVersions(source *v1alpha1.ApplicationSource, kubeVersionFromServer string, apiVersionsFromServer []string) (string, []string) {
```
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply controller/state.go:288]: Yep, I think that makes sense on both counts. I dislike expanding the repo-server request object to add overrides fields, but I think in this case it makes sense.

[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply controller/state.go:288]: Actually, won't have to expand the API, we already send the whole source over to the repo server.

**Comment by crenshaw-dev** [comment]
@agaudreault changes made, can you take another look?

File: docs/operator-manual/application.yaml [hunk 2]
```diff
@@ -137,6 +137,16 @@ spec:
...
+      # You can specify the Kubernetes API version to pass to Helm when templating manifests. By default, Argo CD uses
+      # the Kubernetes version of the target cluster. The value must be semver formatted. Do not prefix with `v`.
+      kubeVersion: 1.30.0
+
+      # You can specify the Kubernetes resource API versions to pass to Helm when templating manifests. By default, Argo
+      # CD uses the API versions of the target cluster. The format is [group/]version/kind.
+      apiVersions:
+        - traefik.io/v1alpha1/TLSOption
+        - v1/Service
+        -
```
Comment (agaudreault) [inline docs/operator-manual/application.yaml:149]: Typo?
```suggestion
      apiVersions:
        - traefik.io/v1alpha1/TLSOption
        - v1/Service

```

**Review by agaudreault — APPROVED** [review]

**Pull request merged**